spatialmath/quaternion.go

```diff
@@ -51,7 +51,15 @@ func QuatToEulerAngles(q quat.Number) *EulerAngles {
 
 	// yaw (z-axis rotation)
 	sinyCosp := 2.0 * (q.Real*q.Kmag + q.Imag*q.Jmag)
+	if math.Abs(sinyCosp) < 1e-16 { // && runtime.GOARCH == "arm64"
```

Comment:
I've had the most success in test cases when this is 1e-12, the commented out bit is also an option but I'd rather we avoid differences because of OS in all cases, but would like to hear your thoughts.

Reply:
particularly - what epsilon is too big here?

Reply:
For the purposes of RDK, we consider two orientations identical when the R4AA angle between them is less than 1e-4 radians.

The values in question here result in angle changes far below that threshold, so I don't have a strong opinion here.

Reply:
cool, I'll change it to the sweet spot I found then.